src/pm/xbps.rs

```diff
@@ -48,14 +49,64 @@ impl Pm for Xbps {
 
     /// Q generates a list of installed packages.
     async fn q(&self, kws: &[&str], flags: &[&str]) -> Result<()> {
-        self.run(Cmd::new(&["xbps-query", "-l"]).kws(kws).flags(flags))
-            .await
+        if kws.is_empty() {
+            return self
+                .run(Cmd::new(&["xbps-query", "-l"]).kws(kws).flags(flags))
+                .await;
+        }
+
+        let results = stream::iter(kws)
+            .filter_map(|&pkg| async {
+                let cmd = Cmd::new(&["xbps-query", "--property", "pkgver", pkg]).flags(flags);
+                self.check_output(cmd, PmMode::Mute, &Strategy::default())
+                    .await
+                    .map(|res| String::from_utf8(res).unwrap())
+                    .ok()
+            })
+            .collect::<Vec<_>>()
+            .await;
+
+        for res in results {
+            println!("{}", res.trim());
+        }
+
+        Ok(())
     }
 
     /// Qe lists packages installed explicitly (not as dependencies).
     async fn qe(&self, kws: &[&str], flags: &[&str]) -> Result<()> {
-        self.run(Cmd::new(&["xbps-query", "-m"]).kws(kws).flags(flags))
-            .await
+        if kws.is_empty() {
+            return self
+                .run(Cmd::new(&["xbps-query", "-m"]).kws(kws).flags(flags))
+                .await;
+        }
+
+        let results = stream::iter(kws)
+            .filter(|&&pkg| async {
+                let check_cmd =
+                    Cmd::new(&["xbps-query", "--property", "automatic-install", pkg]).flags(flags);
+                let result = self
+                    .check_output(check_cmd, PmMode::Mute, &Strategy::default())
+                    .await;
+
+                // if a package is manually installed then the automatic-install field is empty
+                matches!(result, Ok(automatic) if automatic.is_empty())
+            })
+            .filter_map(|&pkg| async {
+                let cmd = Cmd::new(&["xbps-query", "--property", "pkgver", pkg]).flags(flags);
+                self.check_output(cmd, PmMode::Mute, &Strategy::default())
+                    .await
+                    .map(|res| String::from_utf8(res).unwrap())
```

Comment:
Same problem as `async fn q`. You might want to extract a common method here for the `stream`ing part for those two methods, like I did with homebrew here:

https://github.com/rami3l/pacaptr/blob/10689a4709d14dccf3ce712398585bcfd57eb2a1/src/pm/brew.rs#L41-L42